source/common/http/conn_pool_grid.cc

```diff
@@ -236,10 +239,10 @@ ConnectivityGrid::ConnectivityGrid(
 ConnectivityGrid::~ConnectivityGrid() {
   // Ignore idle callbacks while the pools are destroyed below.
   destroying_ = true;
-  // Callbacks might have pending streams registered with the pools, so cancel and delete
-  // the callback before deleting the pools.
-  wrapped_callbacks_.clear();
+  // Make sure to clear the pools (which deletes all active connections, and
+  // signals callers) before deleting callbacks.
```

Comment:
Interesting. So this means we'll end up invoking these callbacks during grid destruction?